services/app/src/components/avatarUploadModal/AvatarUploadModal.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Modal } from 'antd';
+import './avatarModal.css';
+import AvatarModalForm from './AvatarModalForm';
+import AvatarModalComplete from './AvatarModalComplete';
+import { useUserContext } from '../../contexts/UserContext';
+import { buildPlayerAvatarUpdateMessage } from '../../api/playerApi';
+import { useWsContext } from '../../contexts/ws-context';
+
+const AvatarUploadModal = ({ showAvatarModal, closeAvatarModal }) => {
+  const { ws } = useWsContext();
+  const userContext = useUserContext();
+ 
+  const [modalStep, setModalStep] = useState(1);
+  const [imageUrl, setImageUrl] = useState('');
+
+  const onAvatarModalCancel = () => {
+    setModalStep(1);
+    closeAvatarModal();
+    setImageUrl('');
+    userContext.setAvatarError(false);
```

Comment:
do we really need to move avatar error into user context? It seems like something that's localized within this component's scope and should be a state here.